lib/ansible/modules/files/yedit.py

```diff
@@ -0,0 +1,965 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# pylint: disable=wrong-import-order,wrong-import-position,unused-import
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
+
+DOCUMENTATION = '''
+---
+module: yedit
```

Comment:
Bike shed: would be good if this included `yaml` in its name.

Reply:
yedit was chosen for yaml - edit.  It also handles json since it is a subset of yaml.  Basically modifying structured yaml/json.